inception/inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndex.java

```diff
@@ -396,6 +402,221 @@ public long numberOfQueryResults(SearchQueryRequest aRequest)
         return _executeQuery(this::doCountResults, aRequest);
     }
 
+    @Override
+    public StatisticsResult getAnnotationStatistics(StatisticRequest aStatisticRequest)
+        throws IOException, ExecutionException
+    {
+        ArrayList<Integer> fullDocSet = getUniqueDocuments(aStatisticRequest);
+        Map<String, Map<String, Double>> allStats = new HashMap<String, Map<String, Double>>();
```

Comment:
Do we know which metrics we have? If yes, we could use an object here that wraps all these metrics instead of using a Map<String, Double>

Reply:
I wanted to make the code flexible, so the user can decide which metrics should be calculated. Alternatively, we can always calculate all metrics but this could be problematic for the speed.